src/Main.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+
+import Sanitize
+import Scraping
+import Screenshot
+
+from Quiz import Quiz
+from argparse import ArgumentParser
+from colorama import Fore, Back, Style
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+def play(num_questions,path_directory,path_screenshot):
+    # Create a new Quiz
+    quiz = Quiz(path_directory)
+    # Create a ThreadPool to parallelize the work from here on
+    pool = ThreadPoolExecutor(max_workers=4)
+
+    # For each question
+    for i in range(1,num_questions+1):
+        # Waiting for user input...
+        try:
+            c = input("Press enter to evaluate a new question, e to exit: ")
+        except KeyboardInterrupt:
+            # If CTRL+C, break
+            print("") # Go to a new line  
+            break
+        # If the user decided to exit, break
+        if c == 'e': break
+
+        # If we are here to evaluate only one question
+        if path_screenshot:
+            # Load the screenshot from disk as cv2 grey object
+            screenshot = Screenshot.load_image(path_screenshot)
+        # Or we have to process an entire directory of questions
+        elif path_directory:
+            # Go through all screenshots in the specified directory 
+            filename = f"{path_directory}/Question-{i}.png"
+            screenshot = Screenshot.load_image(filename)
+        # If there isn't neither a screenshots nor a directory
+        else:
+            # Define the path for a new screenshot file
+            filename = f"{quiz.folder_name}/Question-{i}.png"
+            # Take a black-n-white screenshot
+            screenshot = Screenshot.take_screenshot(filename)
+
+        # Create a new Question object and extract the question from the screenshot
+        question = quiz.new_question(Screenshot.extract_question(screenshot))
+
+        # Extract all three answers from the screenshot, in three different threads
+        future_answers = {
+            pool.submit(
+                Screenshot.extract_answer,
+                screenshot,
+                question,
+                position
+            ): position for position in range(3)}
+        future_question = pool.submit(Sanitize.clean_question(question))
+        # Wait until all threads are done
+        for future in as_completed(future_answers): pass
```

Comment:
ummm qualcosa mi turba di come hai usato i future qui... question e' usata dal threadpool per accumulare le risposte? Non ci sono problemi di concorrenza? prendi in considerazione di cambiare modello e far restituire i risultati come risutlato e accumularli qui evitando il possibile problema alla radice